compiler/src/dotty/tools/dotc/core/SymDenotations.scala

```diff
@@ -1626,7 +1626,7 @@ object SymDenotations {
           // is defined in the same scope as `cls` or in the companion object of `cls`.
           completeChildrenIn(owner)
           completeChildrenIn(companionClass)
-          setFlag(ChildrenQueried)
+          if isType then setFlag(ChildrenQueried)
```

Comment:
I think it'd be better to skip the whole thing if the symbol is a type, otherwise it will recompute every time

Reply:
I'm not sure what you mean, could you elaborate?

Reply:
So I've come across this again, the problem is calling `children` recursively in `sealedStrictDescendants`, and the `isAllOf(JavaEnumTrait)` will also return true for java enum cases, so best to check that it is also a class